network/connect_wifi.py

```diff
@@ -0,0 +1,52 @@
+import os
+import pexpect
+import sys
+
+size=len(sys.argv)
+passphrase=sys.argv[size-1]
+passphrase.strip(' ')
+ssid=sys.argv[1]
+if size>=4:
+    i=size-2
+    for loop in range(1,i):
+        ssid=ssid+' '+sys.argv[loop+1]
+        print ssid
+
+cmd_arg=', '.join(sys.argv[1:])
+print cmd_arg
+connmanctl=pexpect.spawn('connmanctl')
+connmanctl.expect('connmanctl>')
+connmanctl.sendline('agent on')
+err=connmanctl.expect(['Agent registered', 'Error'])
+if err == 1:
+    print 'Error in agent on'
+f=open('/usr/share/cockpit/network/output.txt','r')
```

Comment:
we should not write "temporary files" on /usr/share. Never.

We should not write temporary file at all for this functionality.... it should communicate with the frontend (js) just like it does in other places. You can open another fd for this, or use stdout/stderr.

Btw, they way you did it, if the client side update while you are writing to the file, you will get partial contents.